systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCorsST.java

```diff
@@ -4,129 +4,109 @@
-                        context.completeNow();
-                    }));
-            }));
+        Map<String, String> additionalHeaders = new HashMap<>();
+        additionalHeaders.put("Origin", NOT_ALLOWED_ORIGIN);
+        additionalHeaders.put("Access-Control-Request-Method", HttpMethod.POST.toString());
+
+        String url = BridgeUtils.DEFAULT_BRIDGE_HOST + "/consumers/" + groupId + "/instances/" + kafkaBridgeUser + "/subscription";
+        String headers = BridgeUtils.addHeadersToString(additionalHeaders);
+        String response = BridgeUtils.executeCurlCommand(HttpMethod.OPTIONS, podName, url, headers);
+
+        LOGGER.info("Checking if response from Bridge is correct");
+        assertThat(response, containsString("403"));
+        assertThat(response, containsString("CORS Rejected - Invalid origin"));
+
+        additionalHeaders.remove("Access-Control-Request-Method", HttpMethod.POST.toString());
+        headers = BridgeUtils.addHeadersToString(additionalHeaders);
+        response = BridgeUtils.executeCurlCommand(HttpMethod.POST, podName, url, headers);
+
+        LOGGER.info("Checking if response from Bridge is correct");
+        assertThat(response, containsString("403"));
+        assertThat(response, containsString("CORS Rejected - Invalid origin"));
     }
 
     @BeforeAll
-    static void beforeAll() throws InterruptedException {
-        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1)
-            .editSpec()
-                .editKafka()
-                    .editListeners()
-                        .withNewKafkaListenerExternalNodePort()
-                            .withTls(false)
-                        .endKafkaListenerExternalNodePort()
-                    .endListeners()
-                .endKafka()
-            .endSpec()
-            .done();
+    void beforeAll() throws Exception {
+        deployClusterOperator(NAMESPACE);
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1).done();
 
         KafkaBridgeResource.kafkaBridgeWithCors(CLUSTER_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME),
-            1, CORS_ORIGIN, null).done();
+            1, ALLOWED_ORIGIN, null).done();
 
         KafkaBridgeHttpCors kafkaBridgeHttpCors = KafkaBridgeResource.kafkaBridgeClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getSpec().getHttp().getCors();
         LOGGER.info("Bridge with the following CORS settings {}", kafkaBridgeHttpCors.toString());
 
-        Service service = KafkaBridgeUtils.createBridgeNodePortService(CLUSTER_NAME, NAMESPACE, bridgeExternalService);
-        KubernetesResource.createServiceResource(service, NAMESPACE).done();
-        ServiceUtils.waitForNodePortService(bridgeExternalService);
-
-        bridgePort = KafkaBridgeUtils.getBridgeNodePort(NAMESPACE, bridgeExternalService);
-        bridgeHost = kubeClient(NAMESPACE).getNodeAddress();
+        podName = kubeClient().listPodsByPrefixInName(CLUSTER_NAME + "-bridge").get(0).getMetadata().getName();
```

Comment:
I think you should be running this in a separate pod. Not from the bridge pod it self.

Reply:
I'm gonna take a look at it :)